### PR TITLE
Add cassandra authentication

### PR DESCRIPTION
**Pull request by jamescway**
- add auth to collector
- add auth to web query

@oscil8 @trane @relwell @rkuris

### CODE REVIEW

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala [hunk 1]
```diff
@@ -254,7 +255,15 @@ class FakeCassandra extends Cassandra.Iface {
     }
   }
 
-  def login(auth_request: AuthenticationRequest) { throw new UnsupportedOperationException }
+  def login(auth_request: AuthenticationRequest) {
+    if(auth_request.credentials("username") == Auth.username &&
```
Comment (relwell) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala]: Looks like you're checking the same thing twice here

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/Cluster.scala [hunk 2]
```diff
@@ -59,12 +59,12 @@ class Cluster(seedHosts: Set[String], seedPort: Int, stats: StatsReceiver, trace
    * Returns a  [[com.twitter.cassie.KeyspaceBuilder]] instance.
    * @param name the keyspace's name
    */
-  def keyspace(name: String): KeyspaceBuilder = {
+  def keyspace(name: String, username: String, password: String): KeyspaceBuilder = {
```
Comment (trane) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/Cluster.scala:62]: what happens to users of this library that don't require authentication? to support the old and new way, you could use method overloading and return a new type that `extends KeyspaceBuilder`, maybe `AuthenticatedKeyspaceBuilder`?

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala [hunk 3]
```diff
@@ -254,7 +255,15 @@ class FakeCassandra extends Cassandra.Iface {
...
+       auth_request.credentials("username") == Auth.username){
```
Comment (trane) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala]: why have an object of `Auth` cluttering the `util.*` namespace? why not use a `val` in this `FakeCassandra` object instead?

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/util/Auth.scala [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.twitter.cassie.util
+import java.nio.ByteBuffer
+
+object Auth {
```
Comment (trane) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/util/Auth.scala]: not sure what you gain here

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/CassieSpanStoreFactory.scala [hunk 5]
```diff
@@ -119,12 +119,16 @@ trait CassieSpanStoreFactory { self: App =>
   val cassieMaxTraceCols = flag("zipkin.store.cassie.maxTraceCols", Defaults.MaxTraceCols, "max number of spans to return from a query")
   val cassieReadBatchSize = flag("zipkin.store.cassie.readBatchSize", Defaults.ReadBatchSize, "max number of rows per query")
 
+  val cassieUsername = flag("zipkin.store.cassie.username", "", "cassandra username")
```
Comment (trane) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/CassieSpanStoreFactory.scala]: i think these user/pass are per-keyspace not per-cassandra, seems weird to call this "cassandra username".

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala [hunk 6]
```diff
@@ -24,31 +24,34 @@ import com.twitter.util.Duration
 import java.net.InetSocketAddress
 
 object Keyspace {
-  def zookeeperServerSets(
```
Comment (trane) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala:27]: why remove this code, why not create a new method that does the not-zookeeper server sets?

File: zipkin-cassandra/src/test/scala/com/twitter/zipkin/storage/cassandra/CassieSpanStoreTest.scala [hunk 7]
```diff
@@ -29,15 +37,38 @@ class CassieSpanStoreTest extends FunSuite {
...
+                         Auth.password
+                         ))
 
   def newSpanStore = {
     FakeServer.reset()
     CassieStore.newCassandraStore()
   }
 
+  // Gets non-visible class variables
+  def get_privates[T](obj: T, fieldname: String) = {
```
Comment (trane) [inline zipkin-cassandra/src/test/scala/com/twitter/zipkin/storage/cassandra/CassieSpanStoreTest.scala:46]: ewww

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala [hunk 8]
```diff
@@ -24,31 +24,34 @@ import com.twitter.util.Duration
...
-    keyspaceName: String = "Zipkin",
-    hosts: Seq[(String, Int)],
-    path: String,
-    timeout: Duration,
-    stats: StatsReceiver = NullStatsReceiver): KeyspaceBuilder = {
+  // this code is not used
```
Comment (relwell) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala]: not a big fan of commenting out logic we can always pull back out of version control

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/Cluster.scala (hunk 2 above)
Comment (jamescway) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/Cluster.scala:62]: So if you try to authenticate with cassandra without the auth turned on, it just acts like a pass through.

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala (hunk 1 above)
Comment (jamescway) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala]: good catch

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala (hunk 3 above)
Comment (jamescway) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/tests/util/FakeCassandra.scala]: yeah, I'm moving it into object FakeCassandra.  Just wanted a central location to get the values.

File: zipkin-cassandra/src/main/scala/com/twitter/cassie/util/Auth.scala (hunk 4 above)
Comment (jamescway) [inline zipkin-cassandra/src/main/scala/com/twitter/cassie/util/Auth.scala]: removed

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala (hunk 6 above)
Comment (jamescway) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/cassandra/Keyspace.scala:27]: yeah I think removing this code wasn't a good idea

**Pull request merged**